src/main/java/com/impossibl/postgres/jdbc/PGResultSet.java

```diff
@@ -874,6 +874,9 @@ public Object getObject(int columnIndex, Map<String, Class<?>> map) throws SQLEx
 
     Class<?> targetType = mapGetType(type, map, statement.connection);
 
+    if (statement.connection.isStrictMode() && InputStream.class.equals(targetType))
```

Comment:
At the very least should this not be in mapGetType?

I might advocate for a bolder change.  Does the spec say, or is it reasonable to assume that, that getObject for BINARY/VARBINARY say we must be returning byte[]? If so, I would map type to it's java SQL type (via SQLTypeUtils) and translate it this way in all cases and forego the "strict" mode.